docs/sources/reference/components/otelcol.exporter.kafka.md

```diff
@@ -0,0 +1,323 @@
+different labels.
+
+## Usage
+
+```alloy
+otelcol.exporter.kafka "LABEL" {
+  protocol_version = "PROTOCOL_VERSION"
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name                                       | Type            | Description                                                                         | Default              | Required
+------------------------------------------ | --------------- | ----------------------------------------------------------------------------------- | -------------------- | --------
+`protocol_version`                         | `string`        | Kafka protocol version to use.                                                      |                      | yes
+`brokers`                                  | `list(string)`  | Kafka brokers to connect to.                                                        | `["localhost:9092"]` | no
+`topic`                                    | `string`        | Kafka topic to read from.                                                           |  _See below_         | no
+`topic_from_attribute`                     | `string`        | A resource attribute whose value should be used as the message's topic.             |  `""`                | no
+`encoding`                                 | `string`        | Encoding of payload read from Kafka.                                                | `"otlp_proto"`       | no
+`client_id`                                | `string`        | Consumer client ID to use. The ID will be used for all produce requests.            | `"sarama"`           | no
+`timeout`                                  | `duration`      | The timeout for every attempt to send data to the backend.                          | `"5s"`               | no
+`resolve_canonical_bootstrap_servers_only` | `bool`          | Whether to resolve then reverse-lookup broker IPs during startup.                   | `"false"`            | no
+`partition_traces_by_id`                   | `bool`          | Whether to include the trace ID as the message key in trace messages sent to Kafka. | `"false"`            | no
+`partition_metrics_by_resource_attributes` | `bool`          | Whether to include the hash of sorted resource attributes as the message partitioning key in metric messages sent to Kafka. | `"false"`            | no
+
+If `topic` is not set, different topics will be used for different telemetry signals:
+
+* Metrics will be received from an `otlp_metrics` topic.
+* Traces will be received from an `otlp_spans` topic.
+* Logs will be received from an `otlp_logs` topic.
+
+If `topic` is set to a specific value, then only the signal type that corresponds to the data stored in the topic must be set in the output block.
+For example, if `topic` is set to `"my_telemetry"`, then the `"my_telemetry"` topic can only contain either metrics, logs, or traces. 
+If it contains only metrics, then `otelcol.exporter.kafka` should be configured to process only metrics.
```

Comment:
if we give a custom topic, will we still start the three types of exporters? If so, can this create problems like we had for the receiver?

Reply:
Apologies, well noted - I'm going to remove this whole paragraph. I'm pretty sure `otelcol.exporter.kafka` has no trouble exporting all signals to the same topic. It's only an issue on the receiver side.